public/app/core/components/RolePicker/TeamRolePicker.tsx

```diff
@@ -5,35 +5,51 @@ import { contextSrv } from 'app/core/core';
 import { Role, AccessControlAction } from 'app/types';
 
 import { RolePicker } from './RolePicker';
-// @ts-ignore
 import { fetchTeamRoles, updateTeamRoles } from './api';
 
 export interface Props {
   teamId: number;
   orgId?: number;
   roleOptions: Role[];
   disabled?: boolean;
-  builtinRolesDisabled?: boolean;
+  onApplyRoles?: (newRoles: Role[]) => void;
+  pendingRoles?: Role[];
+  apply?: boolean;
```

Comment:
`apply` setting refers to whether the RolePicker should send the update request to the server as soon as the RolePicker submenu has been closed. 
* In case of  `apply=false`, RolePicker initiates an update request to the server after the submenu has been closed (User clicks on the `Update` button). 
* In case of `apply=true`, RolePicker updates the current state (in this particular case by calling `onApplyRoles`)  and does not send a request automatically with the updated Roles

Reply:
Hmm, this doesn't feel like a good design choice to me, but I don't have enough knowledge about the components' usage to suggest a better alternative. Maybe we could add some comment to the prop at least, to clarify what it's used for?